src/easier68k/core/models/assembly_parameter.py

```diff
@@ -0,0 +1,219 @@
+"""
+Assembly Parameter
+
+Represents a single parameter in assembly code
+this is the new 'EAMode' class
+"""
+
+from ..enum.ea_mode import EAMode
+from ..enum.register import Register
+from ...simulator.m68k import M68K
+from ..util.conversions import to_word
+
+# should try to make this a constant only defined once
+MAX_MEMORY_LOCATION = 16777216  # 2^24
+
+
+class AssemblyParameter:
+
+    def __init__(self, mode: EAMode = EAMode.ERR, data: int = 0, length: int = 4):
```

Comment:
Do we really want to allow defaults here? They do make sense for when it's an error, but never any other time I can think of.

Also, do we need ERR? Why not return None for errors instead of an AssemblyParameter set to ERR?

Reply:
Both of those are a good point, I agree.